core/src/main/java/io/grpc/Attributes.java

```diff
@@ -189,7 +189,11 @@ public boolean equals(Object o) {
    */
   @Override
   public int hashCode() {
-    return data.hashCode();
+    int hashCode = 0;
+    for (Object o : data.values()) {
+      hashCode += o.hashCode();
```

Comment:
It seems you should include the key hash code as well.

Reply:
Done.